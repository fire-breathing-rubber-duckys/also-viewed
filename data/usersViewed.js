const usersViewed = [
  {
    productId: 1,
    photoURL: 'https://www.rei.com/media/product/148949?size=280',
    brandName: 'Manduka',
    itemName: 'Lean Cork Block',
    itemRating: 5,
    itemPrice: '$16.00',
    ratingCount: 2,
  },
  {
    productId: 2,
    photoURL: 'https://www.rei.com/media/product/182895?size=280',
    brandName: 'Manduka',
    itemName: 'Rectangular Bolster',
    itemRating: 3,
    itemPrice: '$68.00',
    ratingCount: 2,
  },
  {
    productId: 3,
    photoURL: 'https://www.rei.com/media/product/173243?size=280',
    brandName: 'Manduka',
    itemName: 'unBLOK Recycled Foam Yoga Block',
    itemRating: 5,
    itemPrice: '$24.00',
    ratingCount: 3,
  },
  {
    productId: 4,
    photoURL: 'https://www.rei.com/media/product/101409?size=280',
    brandName: 'Jade',
    itemName: 'Cork Yoga Block',
    itemRating: 4.5,
    itemPrice: '$14.95-$19.95',
    ratingCount: 6,
  },
  {
    productId: 5,
    photoURL: 'https://www.rei.com/media/product/182894?size=280',
    brandName: 'Manduka',
    itemName: 'Lean Bolster',
    itemRating: 2,
    itemPrice: '$43.95 $56.00',
    ratingCount: 1,
  },
  {
    productId: 6,
    photoURL: 'https://www.rei.com/media/product/843154?size=280',
    brandName: 'Manduka',
    itemName: 'Recycled Foam Yoga Block',
    itemRating: 5,
    itemPrice: '$16.00',
    ratingCount: 8,
  },
  {
    productId: 7,
    photoURL: 'https://www.rei.com/media/product/850808?size=280',
    brandName: 'Manduka',
    itemName: 'PRO Yoga Mat - Black Sage',
    itemRating: 5,
    itemPrice: '$120.00',
    ratingCount: 18,
  },
  {
    productId: 8,
    photoURL: 'https://www.rei.com/media/product/112757?size=280',
    brandName: 'Manduka',
    itemName: 'Go Move Yoga Mat Sling',
    itemRating: 3,
    itemPrice: '$20.00',
    ratingCount: 77,
  },
  {
    productId: 9,
    photoURL: 'https://www.rei.com/media/product/786833?size=280',
    brandName: 'Jade',
    itemName: 'Harmony Professional Yoga Mat',
    itemRating: 4,
    itemPrice: '$79.95',
    ratingCount: 48,
  },
  {
    productId: 10,
    photoURL: 'https://www.rei.com/media/product/860616?size=280',
    brandName: 'Pro-Tec Athletics',
    itemName: 'Exercise Stretch Band',
    itemRating: 0,
    itemPrice: '$24.95 $29.99',
    ratingCount: 0,
  },
  {
    productId: 11,
    photoURL: 'https://www.rei.com/media/product/120819?size=280',
    brandName: 'Manduka',
    itemName: 'X Yoga Mat',
    itemRating: 4.5,
    itemPrice: '$58.00',
    ratingCount: 7,
  },
];

module.exports = usersViewed;
