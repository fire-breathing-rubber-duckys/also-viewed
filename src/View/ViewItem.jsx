/* eslint-disable jsx-a11y/label-has-associated-control */
import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const ItemWrapper = styled.div`
  margin-left: 20px;
  margin-right: 20px;
  margin-bottom: 55px;
`;

const Item = ({ item }) => {
  const stringArr = item.itemName.split(' ');
  const ratingCount = `(${item.ratingCount})`;
  const stringOne = [];
  const stringTwo = [];

  for (let i = 0; i < stringArr.length; i += 1) {
    if (i < 2) {
      stringOne.push(stringArr[i]);
    } else {
      stringTwo.push(stringArr[i]);
    }
  }

  return (
    <>
      <ItemWrapper>
        <img src={item.photoURL} alt="" />
        <div>
          <div className="item-name">
            <div className="item-brand">{item.brandName}</div>
            {stringOne.join(' ')}
            <br />
            {stringTwo.join(' ')}
          </div>
          <label className="item-rating">{item.itemRating}</label>
          <label className="item-rating-count">{ratingCount}</label>
          <div className="item-price">{item.itemPrice}</div>
        </div>
      </ItemWrapper>
    </>
  );
};

Item.propTypes = {
  item: PropTypes.instanceOf(Object).isRequired,
};

export default Item;
