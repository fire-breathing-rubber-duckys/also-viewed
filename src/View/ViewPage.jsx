import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import Item from './ViewItem';

const PageWrapper = styled.div`
  display: flex;
`;

const ViewPage = ({ data, valid }) => (
  <>
    <PageWrapper>
      {data.map((item) => {
        if (valid.includes(item.productId)) {
          return <Item item={item} key={item.productId} />;
        }
        return undefined;
      })}
    </PageWrapper>
  </>
);

ViewPage.propTypes = {
  data: PropTypes.instanceOf(Array).isRequired,
  valid: PropTypes.instanceOf(Array).isRequired,
};

export default ViewPage;
