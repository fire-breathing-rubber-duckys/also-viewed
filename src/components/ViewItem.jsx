import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import StarRating from './StarRating';

const ItemWrapper = styled.div`
  margin-left: 15px;
  margin-right: 15px;
  margin-bottom: 55px;
`;

const ViewItem = ({ item }) => {
  const stringArr = item.itemName.split(' ');
  const ratingCount = ` (${item.ratingCount})`;
  const stringOne = [];
  const stringTwo = [];

  // Logic to filter the itemName to format in a nice fashion //
  if (stringArr[0].length > 8 && stringArr.length === 2) {
    stringOne.push(stringArr[0]);
    stringTwo.push(stringArr[1]);
  } else if (stringArr[0].length > 8 && stringArr.length === 3) {
    stringOne.push(stringArr[0]);
    stringTwo.push(stringArr[1], stringArr[2]);
  } else {
    for (let i = 0; i < stringArr.length; i += 1) {
      if (i < 2) {
        stringOne.push(stringArr[i]);
      } else {
        stringTwo.push(stringArr[i]);
      }
    }
  }

  return (
    <>
      <ItemWrapper>
        <img src={item.photoURL} alt="" />
        <div>
          <div className="item-name">
            <div className="item-brand">{item.brandName}</div>
            {stringOne.join(' ')}
            <br />
            {stringTwo.join(' ')}
          </div>
          <StarRating rating={item.itemRating} ratingCount={ratingCount} />
          <div className="item-price">{item.itemPrice}</div>
        </div>
      </ItemWrapper>
    </>
  );
};

ViewItem.propTypes = {
  item: PropTypes.instanceOf(Object).isRequired,
};

export default ViewItem;
