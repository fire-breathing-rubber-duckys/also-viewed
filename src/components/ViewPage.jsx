import React from 'react';
import styled from 'styled-components';
import { CSSTransitionGroup } from 'react-transition-group';
import PropTypes from 'prop-types';
import ViewItem from './ViewItem';

const PageWrapper = styled.div`
  display: flex;
`;

const ViewPage = ({ data, valid }) => (
  <>
    <CSSTransitionGroup
      transitionName="carousel"
      transitionEnterTimeout={500}
      transitionLeaveTimeout={500}
      component="div"
    >
      <PageWrapper>
        {data.map((item) => {
          if (valid.includes(item.productId)) {
            return <ViewItem item={item} key={item.productId} />;
          }
          return undefined;
        })}
      </PageWrapper>
    </CSSTransitionGroup>
  </>
);

ViewPage.propTypes = {
  data: PropTypes.instanceOf(Array).isRequired,
  valid: PropTypes.instanceOf(Array).isRequired,
};

export default ViewPage;
