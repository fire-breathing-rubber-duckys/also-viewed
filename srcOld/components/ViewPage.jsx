import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import ViewItem from './ViewItem';

const PageWrapper = styled.div`
  display: flex;
`;

const ViewPage = ({ data, valid }) => (
  <>
    <PageWrapper>
      {data.map((item) => {
        if (valid.includes(item.productId)) {
          return <ViewItem item={item} key={item.productId} />;
        }
        return undefined;
      })}
    </PageWrapper>
  </>
);

ViewPage.propTypes = {
  data: PropTypes.instanceOf(Array).isRequired,
  valid: PropTypes.instanceOf(Array).isRequired,
};

export default ViewPage;
