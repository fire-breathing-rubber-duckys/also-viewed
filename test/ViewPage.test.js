import React from 'react';
import { shallow } from 'enzyme';
import { expect as expectChai } from 'chai';
import renderer from 'react-test-renderer';
import ViewPage from '../src/components/ViewPage';
import ViewItem from '../src/components/ViewItem';
import usersViewed from '../data/usersViewed';
const valid = [1, 2, 3, 4, 5, 6];

describe('View Page Rendering', () => {
  it('should render six ViewItems at a time', () => {
    const wrapper = shallow(<ViewPage data={usersViewed} valid={valid}/>);
    expectChai(wrapper.find(ViewItem)).to.be.lengthOf(6);
  });
});

it('Should render correctly', () => {
  const tree = renderer
    .create(<ViewPage data={usersViewed} valid={valid} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
