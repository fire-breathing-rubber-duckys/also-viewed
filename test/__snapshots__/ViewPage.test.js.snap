// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render correctly 1`] = `
<div>
  <div
    className="sc-gsTCUz fiWKJe"
  >
    <div
      className="sc-bdfBwQ fCuAeK"
    >
      <img
        alt=""
        src="https://www.rei.com/media/product/148949?size=280"
      />
      <div>
        <div
          className="item-name"
        >
          <div
            className="item-brand"
          >
            Manduka
          </div>
          Lean Cork
          <br />
          Block
        </div>
        <div>
          <label
            className="item-rating-count"
          >
             (2)
          </label>
        </div>
        <div
          className="item-price"
        >
          $16.00
        </div>
      </div>
    </div>
    <div
      className="sc-bdfBwQ fCuAeK"
    >
      <img
        alt=""
        src="https://www.rei.com/media/product/182895?size=280"
      />
      <div>
        <div
          className="item-name"
        >
          <div
            className="item-brand"
          >
            Manduka
          </div>
          Rectangular
          <br />
          Bolster
        </div>
        <div>
          <label
            className="item-rating-count"
          >
             (2)
          </label>
        </div>
        <div
          className="item-price"
        >
          $68.00
        </div>
      </div>
    </div>
    <div
      className="sc-bdfBwQ fCuAeK"
    >
      <img
        alt=""
        src="https://www.rei.com/media/product/173243?size=280"
      />
      <div>
        <div
          className="item-name"
        >
          <div
            className="item-brand"
          >
            Manduka
          </div>
          unBLOK Recycled
          <br />
          Foam Yoga Block
        </div>
        <div>
          <label
            className="item-rating-count"
          >
             (3)
          </label>
        </div>
        <div
          className="item-price"
        >
          $24.00
        </div>
      </div>
    </div>
    <div
      className="sc-bdfBwQ fCuAeK"
    >
      <img
        alt=""
        src="https://www.rei.com/media/product/101409?size=280"
      />
      <div>
        <div
          className="item-name"
        >
          <div
            className="item-brand"
          >
            Jade
          </div>
          Cork Yoga
          <br />
          Block
        </div>
        <div>
          <label
            className="item-rating-count"
          >
             (6)
          </label>
        </div>
        <div
          className="item-price"
        >
          $14.95-$19.95
        </div>
      </div>
    </div>
    <div
      className="sc-bdfBwQ fCuAeK"
    >
      <img
        alt=""
        src="https://www.rei.com/media/product/182894?size=280"
      />
      <div>
        <div
          className="item-name"
        >
          <div
            className="item-brand"
          >
            Manduka
          </div>
          Lean Bolster
          <br />
          
        </div>
        <div>
          <label
            className="item-rating-count"
          >
             (1)
          </label>
        </div>
        <div
          className="item-price"
        >
          $56.00
        </div>
      </div>
    </div>
    <div
      className="sc-bdfBwQ fCuAeK"
    >
      <img
        alt=""
        src="https://www.rei.com/media/product/843154?size=280"
      />
      <div>
        <div
          className="item-name"
        >
          <div
            className="item-brand"
          >
            Manduka
          </div>
          Recycled Foam
          <br />
          Yoga Block
        </div>
        <div>
          <label
            className="item-rating-count"
          >
             (8)
          </label>
        </div>
        <div
          className="item-price"
        >
          $16.00
        </div>
      </div>
    </div>
  </div>
</div>
`;
